### ISSUE TITLE
Login with apple is not working in iPad 16.4.1 devices

### ISSUE THREAD

**Issue by Sunsiha** [aws-amplify/amplify-flutter#3100]
Sign in with apple is not working in the iPad device 16.4.1 versions of iOS. Am not getting any console error also. But it is working in real device of iPhone 12 with 16.4.1 version. 

Recently I got an app rejection from Appstore due to the following;

Guideline 2.1 - Performance - App Completeness

We discovered one or more bugs in your app. Specifically, Sign in with Apple feature was unresponsive. Please review the details below and complete the next steps. 

Review device details:

- Device type: iPad 
- OS version: iOS 16.4.1

When I tested on real device(iPhone 12, V.16.4.1) it is working. The same app I tried to run in simulator it is not working. It's stuck with that sign in page itself. 

```
  amplify_flutter: ^0.6.1
  amplify_auth_cognito: ^0.6.1

flutter version details
---------------------
Flutter 3.3.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 4f9d92fbbd (9 months ago) • 2022-09-06 17:54:53 -0700
Engine • revision 3efdf03e73
Tools • Dart 2.18.0 • DevTools 2.15.0
```

In xcode console am getting only like this;

2023-05-24 01:28:27.014946+0530 Runner[44967:2612279] [AuthErrorHandler] The operation couldn’t be completed. (Amplify.AuthError error 1.)

**Comment by Sunsiha** [comment]
@Jordan-Nelson getting debug print like the following;

AuthException(message: invalid_request: PreSignUp failed with error NAME_NOT_FOUND., recoverySuggestion: Retry the webUi signIn, underlyingException: The operation couldn’t be completed. (com.amazon.cognito.AWSCognitoAuthErrorDomain error -3000.))

Let me know how to handle the exception? So that I can show some toast message.

**Comment by Jordan-Nelson** [comment]
@Sunsiha Thanks for the detailed info.

If you don't have any blockers to upgrading to v1, would you be able to see if upgrading to v1.0.1 or higher resolves this issue? Please see the upgrade guide here: https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/

v1 is a complete re-write in dart. Regardless of the issue here, we do recommend upgrading when you are able to as we have resolved many issues in this version that would not have been possible in v0.

**Comment by Jordan-Nelson** [comment]
@Sunsiha do you have a pre sign up lambda trigger configured?

**Comment by Sunsiha** [comment]
> @Sunsiha Thanks for the detailed info.
> 
> If you don't have any blockers to upgrading to v1, would you be able to see if upgrading to v1.0.1 or higher resolves this issue? Please see the upgrade guide here: https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/
> 
> v1 is a complete re-write in dart. Regardless of the issue here, we do recommend upgrading when you are able to as we have resolved many issues in this version that would not have been possible in v0.

Right now If I upgrade not sure any code changes I have to do or not. Now have to upload the application asp. That's why. I will upgrade when I have time.

**Comment by Jordan-Nelson** [comment]
Okay. It seems like that pre sign up trigger is failing. What does the pre sign up trigger do? Can you share the code?

**Comment by Sunsiha** [comment]
> Okay. It seems like that pre sign up trigger is failing. What does the pre sign up trigger do? Can you share the code?

Yeah I can handle with this name not found exception. Since in lambda the code is like this;
```
 const email = event.request.userAttributes.email;

            if (!event.request.userAttributes.name) {

                callback("NAME_NOT_FOUND", event);
            }
```

**Comment by Sunsiha** [comment]
@Jordan-Nelson In some cases we don't get email id and name. So we are throwing an exception from lambda. Since for our registration we need these info. If it is not there we are showing them toast message and redirecting to signup page.

**Comment by Jordan-Nelson** [comment]
Okay. I believe the name parameter is only shared on the very first sign in when using sign in with Apple. This is an intentional design by Apple. So if you sign in with Apple on your iPhone, and then later sign with Apple on your iPad, the name will be empty on the iPad.

Are you seeing this on the iPad when you use sign in with an Apple account that has never signed into this app before on any device. Even if the account was deleted later I think you may not get the name from Apple on subsequent logins.

**Comment by Jordan-Nelson** [comment]
@Sunsiha it sounds like you are able to handle this by catching the AuthException. In v1 there should be a more detailed exception thrown as noted in https://github.com/aws-amplify/amplify-flutter/issues/1279. As mentioned above, the missing name would be expected in some scenarios. If the name is always missing on iPad, even for brand new apple sign ups, we can look into this.

Please let me know if there is anything else I can do to help.

**Comment by Sunsiha** [comment]
> @Sunsiha it sounds like you are able to handle this by catching the AuthException. In v1 there should be a more detailed exception thrown as noted in #1279. As mentioned above, the missing name would be expected in some scenarios. If the name is always missing on iPad, even for brand new apple sign ups, we can look into this.
> 
> Please let me know if there is anything else I can do to help.

Hi @Jordan-Nelson Yes, I am handling it now with AuthException and have submitted a new IPA build. Maybe I need to test it with one or more Apple IDs. Only then I can answer you properly on this. Right now, what you said, I am also suspecting.

**Comment by Sunsiha** [comment]
> Okay. Please let me know if handling the AuthException does not handle this.

The app got approved with that AuthException.

**Comment by Jordan-Nelson** [comment]
@Sunsiha - I'm glad to hear that. I will close this issue out then.
